UnlockSession/handler.ts

```diff
@@ -0,0 +1,124 @@
+import { RequiredBodyPayloadMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/middlewares/required_body_payload";
+import {
+  withRequestMiddlewares,
+  wrapRequestHandler
+} from "@pagopa/io-functions-commons/dist/src/utils/request_middleware";
+import {
+  IResponseErrorForbiddenNotAuthorized,
+  IResponseErrorInternal,
+  IResponseSuccessNoContent,
+  ResponseErrorInternal,
+  ResponseSuccessNoContent,
+  getResponseErrorForbiddenNotAuthorized
+} from "@pagopa/ts-commons/lib/responses";
+import * as express from "express";
+
+import * as E from "fp-ts/Either";
+import * as TE from "fp-ts/TaskEither";
+
+import { readableReportSimplified } from "@pagopa/ts-commons/lib/reporters";
+import { flow, pipe } from "fp-ts/lib/function";
+import { UnlockSessionData } from "../generated/definitions/external/UnlockSessionData";
+import { IConfig } from "../utils/config";
+import { verifyUserEligibilityMiddleware } from "../utils/middlewares/user-eligibility-middleware";
+
+import { UnlockCode } from "../generated/definitions/external/UnlockCode";
+import { Client } from "../generated/definitions/fast-login/client";
+import { SpidLevel } from "../utils/enums/SpidLevels";
+import {
+  IHslJwtPayloadExtended,
+  hslJwtValidationMiddleware
+} from "../utils/middlewares/hsl-jwt-validation-middleware";
+
+type IUnlockSessionHandler = (
+  user: IHslJwtPayloadExtended,
+  payload: UnlockSessionData
+) => Promise<
+  | IResponseSuccessNoContent
+  | IResponseErrorForbiddenNotAuthorized
+  | IResponseErrorInternal
+>;
+
+type UnlockSessionClient = Client<"ApiKeyAuth">;
+
+const canUnlock = (
+  user: IHslJwtPayloadExtended,
+  unlock_code: UnlockCode | undefined
+): boolean =>
+  user.spid_level === SpidLevel.L3 ||
+  (user.spid_level === SpidLevel.L2 && unlock_code !== undefined);
+
+export const unlockSessionHandler = (
+  client: UnlockSessionClient
+): IUnlockSessionHandler => (
+  reqJwtData,
+  reqPayload
+): ReturnType<IUnlockSessionHandler> =>
+  pipe(
+    TE.Do,
+    TE.bind("user_data", () => TE.of(reqJwtData)),
+    TE.bind("unlock_code", () => TE.of(reqPayload.unlock_code)),
```

Comment:
This is an optional value, we can use the `Option` object to map this scenario, avoiding to handle undefined values.